src/cfront.c

```diff
@@ -1308,6 +1309,25 @@ int read_body_assignment(char *token, block_t *parent)
     return 0;
 }
 
+int read_global_assignment(char *token)
+{
+    var_t *var = find_global_var(token);
+    if (var) {
+        char buffer[10];
+        /* global initialization must be constant */
+        /* TODO: support rvalue as expression e.g. int a = 3 + 2; */
+        if (lex_peek(T_numeric, buffer))
```

Comment:
Can you reuse the code in function `read_body_assignment`? That is, split some logic/operations into a helper function and then invoke the helper for reading the expressions from assignments.

Reply:
Do I need to do that anyway? The reason I don't like to do that is:
* The only duplicate code in `read_body_assignment` is `find_global_var`
* Future update about global initialization will make more different with `read_body_assignment`. If I merge now, I guess `read_body_assignment` will become harder to read
* Merge control flow into `read_body_assignment` for one line re-use makes `read_body_assignment` more complicated

Reply:
I agree. Let's stick to the reasonably effective changes.